collections/_posts/2023-02-21-Complementing Brightness Constancy with Deep Networks for Optical Flow Prediction.md

```diff
@@ -0,0 +1,146 @@
+
+The authors propose a hybrid model that decomposes the flow estimation process into a physical flow based on the simplified BC hypothesis and a data-driven augmentation to compensate the limitations of the BC model.
+
+# Methods
+## Architecture
+![](/collections/images/COMBO/architecture.jpg)
+
+To extract the features and to estimate the physical flow, $$ \hat{w}_p $$, augmentation flow, $$ \hat{w}_a $$, and uncertaincy map, $$ \hat{\alpha} $$, RAFT[^1] architecture is used:
+<figure>
+<img src="/collections/images/COMBO/raft.jpg">
+<figcaption align = "center"><b>RAFT</b></figcaption>
+</figure>
+
+## Main ideas of RAFT
+- Feature encoder, $$ g_\theta: \mathbb{R}^{H \times W \times 3} \rightarrow \mathbb{R}^{\frac{H}{8} \times \frac{W}{8} \times D}$$: 
+  - Extract features from image $$ t-1 $$ and $$ t $$ using a simple encoder.
+- Context encoder, $$ h_\theta: \mathbb{R}^{H \times W \times 3} \rightarrow \mathbb{R}^{\frac{H}{8} \times \frac{W}{8} \times D}$$:
+  - Extract context features from image $$ t-1 $$.
+- Visual similarity:
+  - Compute correlation between features extracted from image $$ t-1 $$ and $$ t $$. $$ <g_\theta(I_{t-1}), g_\theta(I_{t})> \in \mathbb{R}^{\frac{H}{8} \times \frac{W}{8} \times \frac{H}{8} \times \frac{W}{8}}$$
+  - Output a 4D correlation volume.
+  - Construct a 4-layer correlation pyramid by pooling the last two dimensions of the correlation volume.
+  - Give information about small and large displacements.
+<figure>
+<img src="/collections/images/COMBO/corr_vol.jpg">
+<figcaption align = "center"><b>Correlation pyramid</b></figcaption>
+</figure>
+- Iterative updates using Gated recurrent units (GRUs):
+  - Current flow estimate, $$ f_k $$ (Initialized as 0)
+  - Apply the flow and update the coordinates of each pixel
+  - Retrieve the correlation features from the correlation pyramids:
+    1. For each predicted pixel in $$ I_t $$, define a local grid of $$ 9\times9 $$ using a radius of $$ 4 $$.
+    2. Retrieve the correlations within this neighborhood for each pyramid level to obtain the information about small and large movements.
+    3. Concatenate the retrieved correlation features of each level $$ \in \mathbb{R}^{\frac{H}{8} \times \frac{W}{8} \times 9 \times 9 \times 4}$$
+  - Inject the concatenation of flow, correlation, and context features to the GRU
+  - Predict flow update, $$\Delta f$$
+  - Update the flow estimate, $$ f_k = f_k + \Delta f$$
+  - Upsample the predicted flow since the output flow is computed at 1/8 resolution.
+
+## Innovations
+Instead of predicting only the physical flow, $$ \hat{w}_p $$ using GRUs, the authors use another GRU block to predict the augmentation flow and uncertainty map. 
```

Comment:
Punctuation:
```suggestion
Instead of predicting only the physical flow $$ \hat{w}_p $$ using GRUs, the authors use another GRU block to predict the augmentation flow and uncertainty map. 
```